src/bc_protocol/media_packet.rs

```diff
@@ -0,0 +1,381 @@
+use crate::bc::model::*;
+use crate::bc_protocol::connection::BcSubscription;
+use err_derive::Error;
+use log::trace;
+use log::*;
+use std::cmp::min;
+use std::collections::VecDeque;
+use std::convert::TryInto;
+use std::str;
+use std::time::Duration;
+
+const INVALID_MEDIA_PACKETS: &[MediaDataKind] = &[
+    MediaDataKind::Invalid,
+    MediaDataKind::Continue,
+    MediaDataKind::Unknown,
+];
+
+// This is used as the minium data to pull from the camera
+// When testing the type
+const MAX_HEADER_SIZE: usize = 32;
+
+pub const CHUNK_SIZE: usize = 40000;
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error(display = "Timeout")]
+    Timeout(#[error(source)] std::sync::mpsc::RecvTimeoutError),
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub enum MediaDataKind {
+    VideoDataIframe,
+    VideoDataPframe,
+    AudioDataAac,
+    AudioDataAdpcm,
+    InfoData,
+    Invalid,
+    Continue,
+    Unknown,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct MediaData {
+    pub data: Vec<u8>,
+}
+
+impl MediaData {
+    pub fn body(&self) -> &[u8] {
+        let lower_limit = self.header_size();
+        let upper_limit = self.data_size() + lower_limit;
+        let len = self.len();
+        if !self.complete() {
+            unreachable!(); // An incomplete packet should be discarded not read, panic if we try
+        }
```

Comment:
Since this function is `pub`, we should not assume that the client code will know to uphold this invariant. Is this `unreachable` because this module upholds the invariant? If so, this should be an `assert` (with a comment explaining why it's impossible).

Reply:
I think I will take it out entirly as then we can removed the complete function too

Reply:
Done